### ISSUE TITLE
Improve randomness of winner selection

### ISSUE THREAD

**Issue by dgw** [dgw/sopel-duel#12]
Statistics tracked by the module in my installation indicate that the winner selection might have a "slight" bias:

```
[07:38:55] <~dgw> ;duels dgw
[07:38:55] <&Kaede> dgw has won 213 out of 486 duels (43.83%).
[07:38:58] <~dgw> ;duels Razkuba
[07:38:59] <&Kaede> Razkuba has won 104 out of 168 duels (61.90%).
```

(Razkuba duels me a lot, and I duel him a lot, so these are relatively significant statistics.)

Idea: Sort combatants list before shuffling, then shuffle it twice instead of once before popping choices off.

Alternate idea: Use `random.choice()` instead of just shuffling and popping `combatants`.

Might need to experiment to find which method (or another entirely) gives results closest to 50/50.

**Comment by dgw** [comment]
Tested randomness of a single random.shuffle() using the script below. Results first:

```
Kleiner:~ dgw$ python shuffle.py
502
498
Kleiner:~ dgw$ python shuffle.py
536
464
Kleiner:~ dgw$ python shuffle.py
501
499
Kleiner:~ dgw$ python shuffle.py
511
489
Kleiner:~ dgw$ python shuffle.py
500
500
Kleiner:~ dgw$ python shuffle.py
494
506
Kleiner:~ dgw$ python shuffle.py
454
546
Kleiner:~ dgw$ python shuffle.py
461
539
Kleiner:~ dgw$ python shuffle.py
521
479
Kleiner:~ dgw$ python shuffle.py
518
482
Kleiner:~ dgw$ python shuffle.py
493
507
Kleiner:~ dgw$ python shuffle.py
525
475
Kleiner:~ dgw$ python shuffle.py
493
507
```

Sometimes it's quite fair, and sometimes it's way off.

Test script:

``` Python
import random

mewins = youwins = 0

for i in range(0,1000):
    combatants = sorted(['me', 'you'])
    random.shuffle(combatants)
    winner = combatants.pop()
    loser = combatants.pop()
    if winner == 'me':
        mewins += 1
    if winner == 'you':
        youwins += 1

print mewins
print youwins
```

**Comment by dgw** [comment]
Adding a second `random.shuffle(combatants)` didn't change the results in any meaningful way:

```
Kleiner:~ dgw$ python shuffle.py
487
513
Kleiner:~ dgw$ python shuffle.py
497
503
Kleiner:~ dgw$ python shuffle.py
501
499
Kleiner:~ dgw$ python shuffle.py
507
493
Kleiner:~ dgw$ python shuffle.py
497
503
Kleiner:~ dgw$ python shuffle.py
496
504
Kleiner:~ dgw$ python shuffle.py
500
500
Kleiner:~ dgw$ python shuffle.py
525
475
Kleiner:~ dgw$ python shuffle.py
485
515
Kleiner:~ dgw$ python shuffle.py
510
490
Kleiner:~ dgw$ python shuffle.py
514
486
Kleiner:~ dgw$ python shuffle.py
499
501
Kleiner:~ dgw$ python shuffle.py
499
501
Kleiner:~ dgw$ python shuffle.py
504
496
```

**Comment by dgw** [comment]
We'll say this is resolved as of 4aad4fb. The balance is Good Enough™ for now.
